test/settings.js

```diff
@@ -289,28 +295,29 @@ module.exports.tests.peliasStreetAnalyzer = function(test, common) {
       "directionals",
       "icu_folding",
       "remove_ordinals",
-      "trim"
+      "trim",
+      "unique_only_same_position",
+      "notnull"
     ]);
     t.end();
   });
 };
 
 // cycle through all analyzers and ensure the corrsponding token filters are defined
 module.exports.tests.allTokenFiltersPresent = function(test, common) {
-  var ES_INBUILT_FILTERS = [
-    'lowercase', 'icu_folding', 'trim', 'word_delimiter', 'unique'
+  var MANDATORY_FILTERS = [
+    'lowercase', 'icu_folding', 'trim', 'unique_only_same_position', 'notnull'
   ];
   test('all token filters present', function(t) {
     var s = settings();
     for( var analyzerName in s.analysis.analyzer ){
       var analyzer = s.analysis.analyzer[analyzerName];
       if( Array.isArray( analyzer.filter ) ){
-        analyzer.filter.forEach( function( tokenFilterName ){
-          var filterExists = s.analysis.filter.hasOwnProperty( tokenFilterName );
-          if( !filterExists && -1 < ES_INBUILT_FILTERS.indexOf( tokenFilterName ) ){
-            filterExists = true;
-          }
-          t.true( filterExists, 'token filter exists: ' + tokenFilterName );
```

Comment:
wow I can't really figure out what the old code was intending to do, but indeed it's clearly wrong (`hasOwnProperty` on an array?)

The new code looks much better.

Reply:
Agh you know what, I think it was actually doing what it was supposed to, after all, I added the original function back in.

So now there are two tests which:
- Ensure that every `token_filter` which was defined in an analyzer is actually globally defined (a sanity check)
- Ensure that a base set of mandatory token filters are present on every analyzer (an integrity check)

Reply:
oh, that makes a lot mores sense. cool!